sigopt/xgboost/utils.py

```diff
@@ -0,0 +1,99 @@
+import xgboost as xgb
+import inspect
+import json
+
+def get_default_args(func):
+  signature = inspect.signature(func)
+  return {
+    k: v.default
+    for k, v in signature.parameters.items() if v.default is not inspect.Parameter.empty
+  }
+
+train_parameters = ['num_boost_round', 'early_stopping_rounds']
+
+def get_train_defaults():
+  train_default_args = get_default_args(xgb.train)
+  defaults = {k: train_default_args[k] for k in train_parameters}
+  return defaults
+
+def parse_parameter(value):
+  for t in (int, float, str):
+    try:
+      ret = t(value)
+      return ret
+    except (ValueError, TypeError):
+      continue
+  return str(value)
+
+def parse_metrics(value):
+  if isinstance(value, str):
+    return value
+  if isinstance(value, list):
+    if len(value) == 1:
+      return value[0]
+  return str(value)
+
+def get_booster_params(booster):
+  # refer:
+  # https://github.com/dmlc/xgboost/blob/406c70ba0e831babce4855d48793df6924e21cbf/python-package/xgboost/sklearn.py#L493
+
+  '''Get xgboost specific parameters.'''
+  config = json.loads(booster.save_config())
+  stack = [config]
+  internal = {}
+  while stack:
+    obj = stack.pop()
+    for k, v in obj.items():
+      if k == 'metrics':
+        internal[k] = parse_metrics(v)
+      if k.endswith('_param'):
+        for p_k, p_v in v.items():
+          internal[p_k] = p_v
+      elif isinstance(v, dict):
+        stack.append(v)
+
+  params = {}
+  for k, v in internal.items():
+    params[k] = parse_parameter(v)
+
+  wrapper_specific = {
+    'importance_type', 'kwargs', 'missing', 'n_estimators', 'use_label_encoder',
+    'enable_categorical'
+  }
+
+  filtered = {}
+  for k, v in params.items():
+    if k not in wrapper_specific and not callable(v):
+      filtered[k] = v
+  return filtered
```

Comment:
What are these for?

Reply:
I will remove the filter. I copy the code from ski-learn.